remotes/docker/resolver.go

```diff
@@ -285,17 +285,19 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 			if err != nil {
 				if errors.Cause(err) == ErrInvalidAuthorization {
 					err = errors.Wrapf(err, "pull access denied, repository does not exist or may require authorization")
+					log.G(ctx).WithError(err).Info("trying next host")
+					continue
 				}
 				return "", ocispec.Descriptor{}, err
 			}
 			resp.Body.Close() // don't care about body contents.
 
-			if resp.StatusCode > 299 {
-				if resp.StatusCode == http.StatusNotFound {
-					continue
-				}
-				return "", ocispec.Descriptor{}, errors.Errorf("unexpected status code %v: %v", u, resp.Status)
+			if resp.StatusCode > 399 {
```

Comment:
What is the rationale behind this change? My understanding is the client already handled the 3xx codes that it code, if 3xx is seen here, we must error. There are currently no other 3xx errors defined in the spec that we need to handle directly.